### PR TITLE
feat(car-recycling): Adding mileage inputs

### PR DESCRIPTION
**Pull request by birkirkristmunds**
## What

 Adding option for user to enter mileage for the car he is recycling.
 Added fix for the ENV variable 

## Screenshots / Gifs
image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by baering — APPROVED** [review]
Core file LGTM.

File: libs/application/templates/car-recycling/src/lib/dataSchema.ts [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-})
+const Vehicles = z
+  .object({
+    make: z.string(),
+    role: z.string(),
+    color: z.string(),
+    permno: z.string(),
+    mileage: z.string(),
+  })
+  .refine(({ mileage }) => mileage !== '' && mileage !== '0')
```
Comment (veronikasif) [inline libs/application/templates/car-recycling/src/lib/dataSchema.ts:11]: Change dataSchema validation to be able to show the user what is causing the error
```suggestion
const Vehicles = z.object({
  make: z.string(),
  role: z.string(),
  color: z.string(),
  permno: z.string(),
  mileage: z.string().refine((m) => m !== '' && m !== '0'),
})
```

Comment (birkirkristmunds) [reply libs/application/templates/car-recycling/src/lib/dataSchema.ts]: fixed

File: libs/application/templates/car-recycling/src/fields/VehiclesOverview/VehiclesOverview.tsx [hunk 2]
```diff
@@ -179,23 +177,79 @@ const VehiclesOverview: FC<FieldBaseProps> = ({ application, error }) => {
...
+                    formatMessage,
+                  )}
+                  backgroundColor="blue"
+                  placeholder="0"
+                  size="sm"
+                  type="number"
+                  defaultValue={vehicle.mileage}
+                  suffix=" "
+                  thousandSeparator
+                  rightAlign={true}
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/car-recycling/src/fields/VehiclesOverview/VehiclesOverview.tsx]: rightAlign removed

File: libs/application/templates/car-recycling/src/fields/VehiclesOverview/VehiclesOverview.tsx [hunk 3]
```diff
@@ -179,23 +177,79 @@ const VehiclesOverview: FC<FieldBaseProps> = ({ application, error }) => {
...
+                    carRecyclingMessages.cars.mileage,
+                    application,
+                    formatMessage,
+                  )}
+                  backgroundColor="blue"
+                  placeholder="0"
+                  size="sm"
+                  type="number"
+                  defaultValue={vehicle.mileage}
+                  suffix=" "
```
[the remark this answers is not in the document]
Comment (birkirkristmunds) [reply libs/application/templates/car-recycling/src/fields/VehiclesOverview/VehiclesOverview.tsx]: not needed, removed

File: apps/application-system/api/infra/application-system-api.ts [hunk 4]
```diff
@@ -298,6 +298,8 @@ export const serviceSetup = (services: {
         '/k8s/api/ALTHINGI_OMBUDSMAN_XROAD_USERNAME',
       ALTHINGI_OMBUDSMAN_XROAD_PASSWORD:
         '/k8s/api/ALTHINGI_OMBUDSMAN_XROAD_PASSWORD',
+      RECYCLING_FUND_GQL_BASE_PATH:
+        '/k8s/skilavottord-ws/RECYCLING_FUND_GQL_BASE_PATH',
```
Comment (AndesKrrrrrrrrrrr) [inline apps/application-system/api/infra/application-system-api.ts:302]: Is this sensitive? Looks like a config option to me. Please remove from `.secrets`

Comment (birkirkristmunds) [reply apps/application-system/api/infra/application-system-api.ts:302]: RECYCLING_FUND_GQL_BASE_PATH  is the path to the skilavottord-ws GraphQL server.
Car recycling application uses the skilavottord-ws  GraphQL server when the application is sent.

File: apps/skilavottord/ws/src/app/modules/vehicle/vehicleAppSys.resolver.ts [hunk 5]
```diff
@@ -35,7 +35,13 @@ export class VehicleAppSysResolver {
...
     )
     if (!vehicle) {
-      throw new NotFoundException(`User does not have this vehicle`)
+      logger.error(
+        'User ' +
+          user.nationalId +
+          ' does not own this vehicle ' +
+          input.permno,
+      )
+      throw new NotFoundException(`User does not own this vehicle`)
```
Comment (AndesKrrrrrrrrrrr) [inline apps/skilavottord/ws/src/app/modules/vehicle/vehicleAppSys.resolver.ts:44]: Please log nicely with template strings, any useful details in object.
```suggestion
logger.error(`User ${user.nationalId} does not own the requested vehicle`, { permno: input.permno, user })
return null
```

Comment (birkirkristmunds) [reply apps/skilavottord/ws/src/app/modules/vehicle/vehicleAppSys.resolver.ts]: Fixed according to suggestion

**Review by veronikasif — APPROVED** [review]
LGTM

File: apps/application-system/api/infra/application-system-api.ts (hunk 4 above)
Comment (saevarma) [reply apps/application-system/api/infra/application-system-api.ts:302]: This shouldn't be a secret. But this is the path to the skilavottord gql server which is our AWS cluster in another namespace.

@island-is/devops / @AndesKrrrrrrrrrrr can they use AWS service url or do the need to go through public path in ingress?

They probably need to grant some namespace in skilavottord-ws/infra to enable this communication?

Comment (saevarma) [reply apps/application-system/api/infra/application-system-api.ts]: @AndesKrrrrrrrrrrr this should be fixed now. Using the `ref` utility to compute the appropriate svc url for AWS cluster. Also granting namepsace access in `skilavottord-ws`.

File: apps/skilavottord/ws/infra/ws.ts [hunk 6]
```diff
@@ -46,3 +46,8 @@ export const serviceSetup = (): ServiceBuilder<'skilavottord-ws'> =>
         paths: ['/app/skilavottord/api/graphql'],
       },
     })
+    .grantNamespaces(
+      'nginx-ingress-external',
+      'nginx-ingress-internal',
```
Comment (AndesKrrrrrrrrrrr) [inline apps/skilavottord/ws/infra/ws.ts:51]: These should not be needed

Comment (AndesKrrrrrrrrrrr) [inline apps/skilavottord/ws/infra/ws.ts:51]: These should not be needed, only `islandis`.

Otherwise, infra LGTM

**Review by AndesKrrrrrrrrrrr — APPROVED** [review]

**Review by busla — DISMISSED** [review]
Friendly suggestion about nicer infra 🙏🏼 

To prevent typos and to make `infra/*` files more readable, consider setting constants for the service names and other redundant strings.

Something like

```js
const namespace = 'application-system'
const serviceName = `${namespace}-api`
const serviceWorkerName = `${serviceName}-worker`
const dbName = `${serviceName.replace('-', '_')}`

const postgresInfo: PostgresInfo = {
  passwordSecret: '/k8s/application-system/api/DB_PASSWORD', // could this be '/k8s/application-system-api/DB_PASSWORD' ?
  name: dbName,
  username: dbName,
}
```

And for the serviceBuilder instances you could do 
```js

export const serviceSetup =
  (): ServiceBuilder<typeof serviceName> =>
//....

export const workerSetup =
  (): ServiceBuilder<typeof serviceWorkerName> =>
```

**Comment by saevarma** [comment]
> Friendly suggestion about nicer infra 🙏🏼 unrelated to your PR line changes.
> 
> To prevent typos and to make `infra/*` files more readable, consider setting constants for the service names and other redundant strings.
> 
> Something like
> 
> ```js
> const namespace = 'application-system'
> const serviceName = `${namespace}-api`
> const serviceWorkerName = `${serviceName}-worker`
> const dbName = `${serviceName.replace('-', '_')}`
> 
> const postgresInfo: PostgresInfo = {
>   passwordSecret: '/k8s/application-system/api/DB_PASSWORD', // could this be '/k8s/application-system-api/DB_PASSWORD' ?
>   name: dbName,
>   username: dbName,
> }
> ```
> 
> And for the serviceBuilder instances you could do
> 
> ```js
> export const serviceSetup =
>   (): ServiceBuilder<typeof serviceName> =>
> //....
> 
> export const workerSetup =
>   (): ServiceBuilder<typeof serviceWorkerName> =>
> ```

This PR is not the correct place for this comment. As this is not a new infra file, this team is only adding changes related to their application, this comment should rather be posted to the codeowners of the application system, which is @island-is/norda.

**Pull request merged**